### ISSUE TITLE
Kernel.Typespec.translate_typespecs_for_module(set, bag) removes all typespecs from a module

### ISSUE THREAD

**Issue by wende** [elixir-lang/elixir#8927]
### Environment
 Elixir version : **Elixir 1.7.1 - 1.8.1**

### Current behaviour

Calling `Kernel.Typespec.translate_typespecs_for_module` removes all of the typespecs from the module's ETS bag

```elixir
defmodule TranslateTypespecExample do
  @type device() :: :stdio

  {set, bag} = :elixir_module.data_tables(__MODULE__)
  # This call below will remove all of the typespecs
  Kernel.Typespec.translate_typespecs_for_module(set, bag)

  @spec a() :: device() # <- This errors out
  def a(), do: :stdio
end

```
Which results in an compilation error of 
```
== Compilation error in file lib/test.ex ==
** (CompileError) lib/test.ex:8: type device/0 undefined
    (elixir) lib/kernel/typespec.ex:891: Kernel.Typespec.compile_error/2
    (elixir) lib/kernel/typespec.ex:832: Kernel.Typespec.typespec/4
    (elixir) lib/kernel/typespec.ex:930: Kernel.Typespec.fn_args/6
    (elixir) lib/kernel/typespec.ex:370: Kernel.Typespec.translate_spec/8
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (elixir) lib/kernel/typespec.ex:223: Kernel.Typespec.translate_typespecs_for_module/2
```

It's effectively caused by the use of take_typespec function in here:
https://github.com/elixir-lang/elixir/blob/eca7e27ad54bf2463b574869e2a1d73f03f12d37/lib/elixir/lib/kernel/typespec.ex#L170-L176

Which uses `:ets.take` instead of `:ets.lookup` which although works for getting single `@spec` to specifically get and remove the last one that was defined, it doesn't make much sense in the context of a function designed to essentially return information about a module at the compile time.

Reproducible error in minimal environment here:
https://github.com/wende/translate_typespec_for_module_issue/blob/master/lib/test.ex

### Expected behaviour

Calling this function should have no side-effects what-so-ever. Alternatively, if the behaviour is intended perhaps a different function name would be better suited

PS. I'd be more than glad to prepare a PR changing this behaviour in case it gets identified as truly unintentional

**Comment by josevalim** [comment]
Hi @wende! Note this module and function are private, so why is it being called directly? Or is the report made because of how Elixir calls it directly?

To be more precise, all functions in the `:elixir*` module are private.

**Comment by wende** [comment]
Hmm... That's true.
I assumed this was the unofficial replacement for removed `Module.get_attribute(:spec)` which no longer can be used since 1.7
It was introduced as a fix here  
https://github.com/elixir-lang/elixir/pull/8289

/Edit. Oh ok. I assumed that this PR got released with Elixir 1.7.1 while actually it's only available since 1.7.4 hence my confusion of not being able to use Module.get_attribute

Then this function is definitely not needed to be exposed since there actually is a replacement. My bad. Closing and thank you 🙌

**Comment by wende** [comment]
It's a shame though that Kernel.Typespec.translate_spec is no longer available
